File: sic_home/lib/models/device.dart
import 'package:json_annotation/json_annotation.dart';

part 'device.g.dart';

enum DeviceDataType {
  @JsonValue(0)
  integer,
  @JsonValue(1)
  float,
  @JsonValue(2)
  string,
  @JsonValue(3)
  boolean,
}

enum DeviceType {
  @JsonValue(0)
  onoff,
  @JsonValue(1)
  analog,
}

@JsonSerializable()
class Device {
  int id;
  String name;
  DeviceType deviceType;
  String userId;
  int placeId;
  int sectionId;
  int pin;

  Device({
    required this.id,
    required this.name,
    required this.deviceType,
    required this.userId,
    required this.placeId,
    required this.sectionId,
    required this.pin,
  });

  factory Device.fromJson(Map<String, dynamic> json) => _$DeviceFromJson(json);

  Map<String, dynamic> toJson() => _$DeviceToJson(this);
}

File: SIC_Backend/Data/Models/Device.cs
﻿using CloudinaryDotNet.Actions;

namespace SIC_Backend.Data.Models
{
    public enum DeviceDataTypes
    {
        Integer,
        Float,
        String,
        Boolean
    }

    public enum DeviceTypes
    {
        OnOff,
        Analog,
    }

    public class Device
    {
        public int Id { get; set; }
        public required string Name { get; set; }
        public required DeviceTypes DeviceType { get; set; }
        public required string UserId { get; set; }
        public int SectionId { get; set; }
        public int PlaceId { get; set; }
        public int Pin { get; set; }

        public User? User { get; set; }
        public Section? Section { get; set; }
        public Place? Place { get; set; }

        public static Device FromCreateModel(CreateDeviceModel model)
        {
            return new Device
            {
                Name = model.Name,
                DeviceType = model.DeviceType,
                UserId = model.UserId,
                PlaceId = model.PlaceId,
                SectionId = model.SectionId,
            };
        }
    }
}
